### ISSUE TITLE
Export types declaration

### ISSUE THREAD

**Issue by AndreaPontrandolfo** [JamieMason/eslint-plugin-prefer-arrow-functions#44]
## Description

With ESLint V9, it now became very valuable for ESlint plugins to export types declaration.

## Suggested Solution

The project is already in Typescript. Use some kind of bundler to export the types declaration.

**Comment by loucyx** [comment]
@AndreaPontrandolfo I agree with you, as a temporary measure this is what I have in my project and it might help you as well. Create a `plugins.d.ts` in your root (make sure is included in your `tsconfig.json`, and then put this on it:

```typescript
declare module "@missing-types-plugin" {
	import type { FlatConfig } from "@typescript-eslint/utils/ts-eslint";

	/**
	 * @deprecated FIXME: doesn't have types.
	 */
	const missing: FlatConfig.Plugin;

	export default missing;
}

declare module "eslint-plugin-arrow-functions" {
	export { default } from "@missing-types-plugin";
}
```

You can use `@missing-types-plugin` with any plugin that has either missing or bad types for ESLint:

```typescript
declare module "<any plugin name here>" {
	export { default } from "@missing-types-plugin";
}
```

Hope it helps temporarily, but 100% types should be exposed from the plugin itself.

**Comment by JamieMason** [comment]
I've not used ESLint v9 but I've done some work to make more use of `@typescript-eslint` and to export the types in https://github.com/JamieMason/eslint-plugin-prefer-arrow-functions/commit/949b6e384b8fe3d9bc18668bd52857e27e5b935b, hopefully this change improves things.

**Comment by JamieMason** [comment]
(Please reopen if you do return to this and it is still an issue)
